Program.cs

```diff
@@ -30,47 +27,52 @@ public static void Main(params string[] args)
                 Console.WriteLine("Ditto will still work as a local database.");
             }
 
-            Console.WriteLine("Welcome to Ditto's Task App");
+            Console.WriteLine("Welcome to Ditto's Task App\n");
+
+            string query = "SELECT * FROM tasks WHERE isDeleted == false";
+
+            ditto.Sync.RegisterSubscription(query);
 
-            subscription = ditto.Store["tasks"].Find("!isDeleted").Subscribe();
-            liveQuery = ditto.Store["tasks"].Find("!isDeleted").ObserveLocal((docs, _event) => {
-                tasks = docs.ConvertAll(document => new Task(document));
+            ditto.Store.RegisterObserver(query, (result) =>
+            {
+                tasks = result.Items.ConvertAll(item => Task.JsonToTask(item.JsonString()));
             });
 
-            ditto.Store["tasks"].Find("isDeleted == true").Evict();
+            await ditto.Store.ExecuteAsync("EVICT FROM tasks WHERE isDeleted == true");
 
             ListCommands();
 
+
             while (!isAskedToExit)
             {
-                
+
                 Console.Write("\nYour command: ");
                 string command = Console.ReadLine();
 
                 switch (command)
                 {
-                    
-                    case string s when command.StartsWith("--upsert"):
-                        string taskBody = s.Replace("--upsert ", "");
-                        ditto.Store["tasks"].Upsert(new Task(taskBody, false).ToDictionary());
+                    case string s when command.StartsWith("--insert"):
+                        string taskBody = s.Replace("--insert ", "");
+                        var task = new Task(taskBody, false).ToDictionary();
+                        await ditto.Store.ExecuteAsync("INSERT INTO tasks DOCUMENTS (:task)", new Dictionary<string, object> { { "task", task } });
                         break;
                     case string s when command.StartsWith("--toggle"):
                         string _idToToggle = s.Replace("--toggle ", "");
-                        ditto.Store["tasks"]
-                            .FindById(new DittoDocumentId(_idToToggle))
-                            .Update((mutableDoc) => {
-                                if (mutableDoc == null) return;
-                                mutableDoc["isCompleted"].Set(!mutableDoc["isCompleted"].BooleanValue);
-                            });
+                        await ditto.Store.ExecuteAsync("SELECT * FROM tasks WHERE _id == :id", new Dictionary<string, object> { { "id", _idToToggle } }).ContinueWith((result) =>
```

Comment:
Can't we simply run the Update query here? Instead of doing a Select first. We have the tasks collection stored in `tasks`. So something like this (not tested): 

```csharp
var newValue = !tasks.First(t => t._id == _idToToggle).isCompleted; 
await ditto.Store.ExecuteAsync($"UPDATE tasks" +
        $"SET isCompleted = {newValue} " +
        $"WHERE _id = '{_idToToggle}' AND isCompleted != {newValue}");
```